### PR TITLE
gnutls: add v3.8.3

### PR DESCRIPTION
**Pull request by alecbcs**
https://lists.gnupg.org/pipermail/gnutls-help/2024-January/004841.html

### CODE REVIEW

File: var/spack/repos/builtin/packages/gnutls/package.py [hunk 1]
```diff
@@ -36,6 +39,9 @@ class Gnutls(AutotoolsPackage):
     # gnutls+guile is currently broken on MacOS.  See Issue #11668
     conflicts("+guile", when="platform=darwin")
 
+    # newer versions of clang are not compatible with gnutls <= 3.7
+    conflicts("%clang@15:", when="@:3.7")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/gnutls/package.py]: I can't reproduce. Only thing is that it sets a bunch of `-W` warning flags that are gcc only, but that's not an error in clang.

File: var/spack/repos/builtin/packages/gnutls/package.py [hunk 2]
```diff
@@ -36,6 +39,10 @@ class Gnutls(AutotoolsPackage):
     # gnutls+guile is currently broken on MacOS.  See Issue #11668
     conflicts("+guile", when="platform=darwin")
 
+    # newer versions of clang are not compatible with gnutls <= 3.7
```
Comment (haampie) [inline var/spack/repos/builtin/packages/gnutls/package.py]:
```suggestion
    # -Wimplicit-int is an error in newer clang
```

**Review by haampie — APPROVED** [review]

**Comment by haampie** [comment]
@spackbot run pipeline

**Comment by bernhardkaindl** [comment]
Failed because of Runner system failure in jobs. Another pipeline just completed, so:
@spackbot run pipeline

**Review by bernhardkaindl — APPROVED** [review]

**Pull request merged**